From 06dff71a31d124ec4c5b0143887b0ec874a672ad Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Wed, 2 Jan 2013 20:56:59 +0100
Subject: [PATCH] Revert "[configure] Cleanup check for ICU"

This reverts commit b161bfc4f6f2db0edea780b95b798ff7b559cf33.
The check is needed to build on old versions of ICU, such as that
shipped by OpenEmbedded.
---
 configure.ac | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 90ca102..7603471 100644
--- a/configure.ac
+++ b/configure.ac
@@ -169,7 +169,26 @@ AM_CONDITIONAL(HAVE_CAIRO_FT, $have_cairo_ft)
 
 dnl ==========================================================================
 
-PKG_CHECK_MODULES(ICU, icu-uc, have_icu=true, have_icu=false)
+dnl The following check is misleading since it would print ICU...no if there's
+dnl no pkgconfig file for icu.
+PKG_CHECK_MODULES(ICU, icu, have_icu=true, [
+	have_icu=true
+	AC_CHECK_HEADERS(unicode/uchar.h,, have_icu=false)
+	AC_MSG_CHECKING([for libicuuc])
+	LIBS_old=$LIBS
+	LIBS="$LIBS -licuuc"
+	AC_TRY_LINK([#include <unicode/uchar.h>],
+		    [u_getIntPropertyValue (0, (UProperty)0);],
+		    AC_MSG_RESULT(yes),
+		    AC_MSG_RESULT(no);have_icu=false)
+	LIBS=$LIBS_old
+	if $have_icu; then
+		ICU_CFLAGS=-D_REENTRANT
+		ICU_LIBS="-licuuc"
+		AC_SUBST(ICU_CFLAGS)
+		AC_SUBST(ICU_LIBS)
+	fi
+])
 if $have_icu; then
 	AC_DEFINE(HAVE_ICU, 1, [Have ICU library])
 fi
-- 
1.8.0.1

